packages/flutter/lib/src/services/browser_context_menu.dart

```diff
@@ -0,0 +1,72 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+
+import 'system_channels.dart';
+
+/// Controls the browser's context menu on the web platform.
+///
+/// The context menu is the menu that appears on right clicking or selecting
+/// text in the browser, for example.
+///
+/// On web, by default, the browser's context menu is enabled by default, and
+/// Flutter's context menus are hidden.
+///
+/// On all non-web platforms, this does nothing.
+class BrowserContextMenu {
+  BrowserContextMenu._();
+
+  static final BrowserContextMenu _instance = BrowserContextMenu._();
+
+  /// Whether showing the browser's context menu is enabled.
+  ///
+  /// When true, any event that the browser typically uses to trigger its
+  /// context menu (e.g. right click) will do so. When false, the browser's
+  /// context menu will not show.
+  ///
+  /// It's possible for this to be true but for the browser's context menu to
+  /// not show due to direct manipulation of the DOM. This bool only indicates
```

Comment:
could you elaborate on what "direct manipulation of the DOM" means? Therer'e ways for the developer to change the underlying DOM element so it's not guaranteed the web menu will be there even if it's enabled?

Reply:
Also if you call `disable/enableContenxtMenu` it won't take effect immediately, and enable reflect the "real" state,  maybe that's worth documenting.

Reply:
For example, if you paste this into your browser's console:

```javascript
document.body.addEventListener('contextmenu', (event) => event.preventDefault());
```

Then Flutter won't know about it and BrowserContextMenu.enabled will still be true even though the context menu won't show.  This could be done at any point in the DOM tree between the event and the root, from anywhere in the page's JS.

Reply:
Good point on your second comment, I'll clarify in the docs.

I'll also add my example in my previous comment to the docs to make it more clear what I mean.